src/Particles/gpuParticleSystem.ts

```diff
@@ -733,11 +765,18 @@ export class GPUParticleSystem extends BaseParticleSystem implements IDisposable
             this.defaultProjectionMatrix = Matrix.PerspectiveFovLH(0.8, 1, 0.1, 100);
         }
 
-        this._customEffect = { 0: customEffect };
+        this._platform = this._engine.getCaps().supportComputeShaders ? new ComputeShaderParticleSystem(this, this._engine) : new WebGL2ParticleSystem(this, this._engine);

```

Comment:
I would love that this code to be delegated nsomehow to the children

My ultimate goal is for someone referencing the GPU particle system for webgl2 to NOT have to import webgpu one

Reply:
I tried something, let me know if it's ok for you.